Assets/Scripts/UI/Indicator/TargetIndicator.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class TargetIndicator : MonoBehaviour
+{
+    public Sprite humanIndicator, ghostIndicator;
+    
+    public Image TargetIndicatorImage;
+    public Image OffScreenTargetIndicator;
+    public float OutOfSightOffset = 20f;
+    private float outOfSightOffest { get { return OutOfSightOffset /* canvasRect.localScale.x*/; } }
+
+    private GameObject target;
+    private Camera mainCamera;
+    private RectTransform canvasRect;
+
+    private RectTransform rectTransform;
+
+    private void Awake()
+    {
+        rectTransform = GetComponent<RectTransform>();
+    }
+
+    private void Start()
+    {
+        if (target.GetComponent<TargetObject>().thisProfile == TargetObject.Profile.Ghost)
+        {
+            OffScreenTargetIndicator.sprite = ghostIndicator;
+        }
+        else
+        {
+            OffScreenTargetIndicator.sprite = humanIndicator;
+        }
+    }
+
+
+    public void InitialiseTargetIndicator(GameObject target, Camera mainCamera, Canvas canvas)
+    {
+        this.target = target;
+        this.mainCamera = mainCamera;
+        canvasRect = canvas.GetComponent<RectTransform>();
```

Comment:
I like to do all my references in Awake() then logic and behaviours in Start() and afterwards, unless this needs to be done in runtime ofc.